### ISSUE TITLE
[EDL]单机多卡fleet训练提示Nccl error异常

### ISSUE THREAD

**Issue by chenzaiping-dev** [PaddlePaddle/Paddle#24809]
本地环境：
Tesla P4 卡
paddlepaddle-gpu==1.7.0.post107
cuda10.1，
cudnn_v7.5
nccl2.4.2
gcc82

跑的训练脚本见:
https://github.com/elasticdeeplearning/edl/blob/develop/example/distill/mnist_distill/train_student_with_fleet.py#L127-L144

运行函数栈见下面：
-------------------------------
API is deprecated since 2.0.0 Please use FleetAPI instead.
WIKI: https://github.com/PaddlePaddle/Fleet/blob/develop/markdown_doc/transpiler

        
W0529 18:26:58.303560 30147 device_context.cc:237] Please NOTE: device: 0, CUDA Capability: 61, Driver API Version: 10.1, Runtime API Version: 10.0
W0529 18:26:58.306265 30147 device_context.cc:245] device: 0, cuDNN Version: 6.0.
W0529 18:26:58.306282 30147 device_context.cc:271] WARNING: device: 0. The installed Paddle is compiled with CUDNN 7.6, but CUDNN version in your machine is 6.0, which may cause serious incompatible bug. Please recompile or reinstall Paddle with compatible CUDNN version.
I0529 18:26:59.316160 30147 rpc_client.h:107] init rpc client with trainer_id 0
[INFO 2020-05-29 18:27:05,378 distill_reader.py:231] Adding server=10.255.100.13:9293
[INFO 2020-05-29 18:27:05,379 distill_worker.py:219] connect server=10.255.100.13:9293
[INFO 2020-05-29 18:27:05,379 distill_worker.py:77] (u'10.255.100.13:9293', './serving_conf/serving_client_conf.prototxt', ['img', 'label'], ['prediction'], 3)
WARNING: Logging before InitGoogleLogging() is written to STDERR
I0529 18:27:05.403625 40650 naming_service_thread.cpp:209] brpc::policy::ListNamingService("10.255.100.13:9293"): added 1
I0529 18:27:06.199968 30147 parallel_executor.cc:440] The Program will be executed on CUDA using ParallelExecutor, 4 cards are used, so 4 programs are executed in parallel.
/home/ssd1/tool/python/lib/python2.7/site-packages/paddle/fluid/executor.py:782: UserWarning: The following exception is not an EOF exception.
  "The following exception is not an EOF exception.")
Traceback (most recent call last):
  File "train_student_with_fleet.py", line 295, in <module>
    main(use_cuda=use_cuda, nn_type=predict)
  File "train_student_with_fleet.py", line 277, in main
    params_filename=params_filename)
  File "train_student_with_fleet.py", line 196, in train
    metrics = exe.run(main_program, feed=data, fetch_list=[loss, acc])
  File "/home/ssd1/tool/python/lib/python2.7/site-packages/paddle/fluid/executor.py", line 783, in run
    six.reraise(*sys.exc_info())
  File "/home/ssd1/tool/python/lib/python2.7/site-packages/paddle/fluid/executor.py", line 778, in run
    use_program_cache=use_program_cache)
  File "/home/ssd1/tool/python/lib/python2.7/site-packages/paddle/fluid/executor.py", line 833, in _run_impl
    program._compile(scope, self.place)
  File "/home/ssd1/tool/python/lib/python2.7/site-packages/paddle/fluid/compiler.py", line 424, in _compile
    places=self._places)
  File "/home/ssd1/tool/python/lib/python2.7/site-packages/paddle/fluid/compiler.py", line 377, in _compile_data_parallel
    self._exec_strategy, self._build_strategy, self._graph)
paddle.fluid.core_avx.EnforceNotMet: 

--------------------------------------------
C++ Call Stacks (More useful to developers):
--------------------------------------------
0   std::string paddle::platform::GetTraceBackString<char const*>(char const*&&, char const*, int)
1   paddle::platform::EnforceNotMet::EnforceNotMet(std::__exception_ptr::exception_ptr, char const*, int)
2   paddle::platform::NCCLContextMap::NCCLContextMap(std::vector<paddle::platform::Place, std::allocator<paddle::platform::Place> > const&, ncclUniqueId*, unsigned long, unsigned long)
3   paddle::platform::NCCLCommunicator::InitFlatCtxs(std::vector<paddle::platform::Place, std::allocator<paddle::platform::Place> > const&, std::vector<ncclUniqueId*, std::allocator<ncclUniqueId*> > const&, unsigned long, unsigned long)
4   paddle::framework::ParallelExecutorPrivate::InitNCCLCtxs(paddle::framework::Scope*, paddle::framework::details::BuildStrategy const&)
5   paddle::framework::ParallelExecutorPrivate::InitOrGetNCCLCommunicator(paddle::framework::Scope*, paddle::framework::details::BuildStrategy*)
6   paddle::framework::ParallelExecutor::ParallelExecutor(std::vector<paddle::platform::Place, std::allocator<paddle::platform::Place> > const&, std::vector<std::string, std::allocator<std::string> > const&, std::string const&, paddle::framework::Scope*, std::vector<paddle::framework::Scope*, std::allocator<paddle::framework::Scope*> > const&, paddle::framework::details::ExecutionStrategy const&, paddle::framework::details::BuildStrategy const&, paddle::framework::ir::Graph*)

----------------------
Error Message Summary:
----------------------
Error: An error occurred here. There is no accurate error hint for this error yet. We are continuously in the process of increasing hint for this kind of error check. It would be helpful if you could inform us of how this conversion went by opening a github issue. And we will resolve it with high priority.
  - New issue link: https://github.com/PaddlePaddle/Paddle/issues/new
  - Recommended issue content: all error stack information
  [unhandled system error] at (/paddle/paddle/fluid/platform/nccl_helper.h:113)

W0529 18:27:06.378182 40650 init.cc:209] Warning: PaddlePaddle catches a failure signal, it may not work properly
W0529 18:27:06.378242 40650 init.cc:211] You could check whether you killed PaddlePaddle thread/process accidentally or report the case to PaddlePaddle
W0529 18:27:06.378249 40650 init.cc:214] The detail failure signal is:

W0529 18:27:06.378258 40650 init.cc:217] *** Aborted at 1590748026 (unix time) try "date -d @1590748026" if you are using GNU date ***
W0529 18:27:06.378240 40642 init.cc:209] Warning: PaddlePaddle catches a failure signal, it may not work properly
W0529 18:27:06.378321 40642 init.cc:211] You could check whether you killed PaddlePaddle thread/process accidentally or report the case to PaddlePaddle
W0529 18:27:06.378326 40642 init.cc:214] The detail failure signal is:

W0529 18:27:06.378338 40642 init.cc:217] *** Aborted at 1590748026 (unix time) try "date -d @1590748026" if you are using GNU date ***
W0529 18:27:06.380712 40650 init.cc:217] PC: @                0x0 (unknown)
W0529 18:27:06.380812 40642 init.cc:217] PC: @                0x0 (unknown)
W0529 18:27:06.380875 40650 init.cc:217] *** SIGTERM (@0x1f4000075c3) received by PID 40650 (TID 0x7f2cd11ed700) from PID 30147; stack trace: ***
W0529 18:27:06.381332 40642 init.cc:217] *** SIGTERM (@0x1f4000075c3) received by PID 40642 (TID 0x7f2cd11ed700) from PID 30147; stack trace: ***
W0529 18:27:06.382815 40650 init.cc:217]     @     0x7f2d46661160 (unknown)
W0529 18:27:06.383354 40642 init.cc:217]     @     0x7f2d46661160 (unknown)
W0529 18:27:06.384951 40650 init.cc:217]     @     0x7f2d45c7cfc9 syscall
W0529 18:27:06.385305 40642 init.cc:217]     @     0x7f2d4665f580 sem_wait
W0529 18:27:06.385643 40650 init.cc:217]     @     0x7f2bf9c1d9b3 bthread::wait_pthread()
W0529 18:27:06.386121 40650 init.cc:217]     @     0x7f2bf9c1e9ec bthread::butex_wait()
W0529 18:27:06.386552 40650 init.cc:217]     @     0x7f2bf9b05d40 bthread_id_join
W0529 18:27:06.386994 40642 init.cc:217]     @     0x7f2d3e25c865 semlock_acquire
W0529 18:27:06.387001 40650 init.cc:217]     @     0x7f2bf9a7e092 brpc::ParallelChannel::CallMethod()
W0529 18:27:06.387137 40642 init.cc:217]     @           0x4b8855 PyEval_EvalFrameEx
W0529 18:27:06.387271 40642 init.cc:217]     @           0x4b87cd PyEval_EvalFrameEx
W0529 18:27:06.387408 40642 init.cc:217]     @           0x4baa48 PyEval_EvalCodeEx
W0529 18:27:06.387425 40650 init.cc:217]     @     0x7f2bf99f4309 baidu::paddle_serving::predictor::general_model::GeneralModelService::CallMethod()
W0529 18:27:06.387509 40642 init.cc:217]     @           0x530e5f function_call
W0529 18:27:06.387650 40642 init.cc:217]     @           0x422fea PyObject_Call
W0529 18:27:06.387775 40642 init.cc:217]     @           0x4b3639 PyEval_EvalFrameEx
W0529 18:27:06.387820 40650 init.cc:217]     @     0x7f2bf9a03af6 baidu::paddle_serving::sdk_cpp::PredictorImpl<>::inference()
W0529 18:27:06.387887 40642 init.cc:217]     @           0x4b87cd PyEval_EvalFrameEx
W0529 18:27:06.388008 40642 init.cc:217]     @           0x4b87cd PyEval_EvalFrameEx
W0529 18:27:06.388059 40650 init.cc:217]     @     0x7f2bf9949da6 baidu::paddle_serving::general_model::PredictorClient::numpy_predict()
W0529 18:27:06.388132 40642 init.cc:217]     @           0x4baa48 PyEval_EvalCodeEx
W0529 18:27:06.388240 40642 init.cc:217]     @           0x530d90 function_call
W0529 18:27:06.388288 40650 init.cc:217]     @     0x7f2bf9953e73 _ZZN8pybind1112cpp_function10initializeIZN5baidu14paddle_serving13general_modelL28pybind11_init_serving_clientERNS_6moduleEEUlRNS4_15PredictorClientERKSt6vectorIS9_INS_7array_tIfLi16EEESaISB_EESaISD_EERKS9_ISsSaISsEERKS9_IS9_IiSaIiEESaISN_EERKS9_IS9_INSA_IlLi16EEESaISS_EESaISU_EESL_SR_SL_RNS4_12PredictorResERKiE12_iIS8_SH_SL_SR_SY_SL_SR_SL_S10_S12_EINS_4nameENS_9is_methodENS_7siblingENS_10call_guardIINS_18gil_scoped_releaseEEEEEEEvOT_PFT0_DpT1_EDpRKT2_ENUlRNS_6detail13function_callEE1_4_FUNES1N_
W0529 18:27:06.388365 40642 init.cc:217]     @           0x422fea PyObject_Call
W0529 18:27:06.388424 40642 init.cc:217]     @           0x4274bd instancemethod_call
W0529 18:27:06.388536 40650 init.cc:217]     @     0x7f2bf99640a9 pybind11::cpp_function::dispatcher()
W0529 18:27:06.388556 40642 init.cc:217]     @           0x422fea PyObject_Call
W0529 18:27:06.388635 40642 init.cc:217]     @           0x48444f slot_tp_init
W0529 18:27:06.388702 40650 init.cc:217]     @           0x4b8b27 PyEval_EvalFrameEx
W0529 18:27:06.388706 40642 init.cc:217]     @           0x47f43a type_call
W0529 18:27:06.388875 40642 init.cc:217]     @           0x422fea PyObject_Call
W0529 18:27:06.388882 40650 init.cc:217]     @           0x4baa48 PyEval_EvalCodeEx
W0529 18:27:06.389024 40642 init.cc:217]     @           0x4b2cfa PyEval_EvalFrameEx
W0529 18:27:06.389039 40650 init.cc:217]     @           0x4b82a8 PyEval_EvalFrameEx
W0529 18:27:06.389173 40642 init.cc:217]     @           0x4b87cd PyEval_EvalFrameEx
W0529 18:27:06.389191 40650 init.cc:217]     @           0x4b87cd PyEval_EvalFrameEx
W0529 18:27:06.389317 40642 init.cc:217]     @           0x4b87cd PyEval_EvalFrameEx
W0529 18:27:06.389340 40650 init.cc:217]     @           0x4b87cd PyEval_EvalFrameEx
W0529 18:27:06.389434 40642 init.cc:217]     @           0x52e6aa gen_iternext
W0529 18:27:06.389477 40650 init.cc:217]     @           0x4b87cd PyEval_EvalFrameEx
W0529 18:27:06.389562 40642 init.cc:217]     @           0x4b5a3c PyEval_EvalFrameEx
W0529 18:27:06.389628 40650 init.cc:217]     @           0x4baa48 PyEval_EvalCodeEx
W0529 18:27:06.389672 40642 init.cc:217]     @           0x52e6aa gen_iternext
W0529 18:27:06.389747 40650 init.cc:217]     @           0x530e5f function_call
W0529 18:27:06.389803 40642 init.cc:217]     @           0x4b5a3c PyEval_EvalFrameEx
W0529 18:27:06.389900 40650 init.cc:217]     @           0x422fea PyObject_Call
W0529 18:27:06.389920 40642 init.cc:217]     @           0x52e6aa gen_iternext
W0529 18:27:06.390048 40650 init.cc:217]     @           0x4b3639 PyEval_EvalFrameEx
W0529 18:27:06.390065 40642 init.cc:217]     @           0x4b5a3c PyEval_EvalFrameEx
W0529 18:27:06.390194 40650 init.cc:217]     @           0x4b87cd PyEval_EvalFrameEx
W0529 18:27:06.390252 40642 init.cc:217]     @           0x4baa48 PyEval_EvalCodeEx
W0529 18:27:06.390337 40650 init.cc:217]     @           0x4b87cd PyEval_EvalFrameEx
W0529 18:27:06.390372 40642 init.cc:217]     @           0x530e5f function_call
W0529 18:27:06.390487 40650 init.cc:217]     @           0x4baa48 PyEval_EvalCodeEx
W0529 18:27:06.390528 40642 init.cc:217]     @           0x422fea PyObject_Call
W0529 18:27:06.390601 40650 init.cc:217]     @           0x530d90 function_call
W0529 18:27:06.390662 40642 init.cc:217]     @           0x4b3639 PyEval_EvalFrameEx
W0529 18:27:06.390739 40650 init.cc:217]     @           0x422fea PyObject_Call
W0529 18:27:06.390784 40650 init.cc:217]     @           0x4274bd instancemethod_call
W0529 18:27:06.390905 40650 init.cc:217]     @           0x422fea PyObject_Call
W0529 18:27:06.390972 40650 init.cc:217]     @           0x48444f slot_tp_init
W0529 18:27:06.391034 40650 init.cc:217]     @           0x47f43a type_call
W0529 18:27:06.391156 40650 init.cc:217]     @           0x422fea PyObject_Call
W0529 18:27:06.391273 40650 init.cc:217]     @           0x4b2cfa PyEval_EvalFrameEx

**Comment by chenzaiping-dev** [comment]
CentOS release 6.3 
2.6内核

**Comment by gongweibao** [comment]
Resolve on Hi
